src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -133,7 +133,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Throw ExceptionUtilities.UnexpectedValue(Me.MemberSymbol.Kind)
             End Select
 
-            Return boundInitializer
+            Dim expressionInitializer = TryCast(boundInitializer, BoundExpression)
+            Return If(expressionInitializer IsNot Nothing AndAlso expressionInitializer.Kind <> BoundKind.BadExpression, New BoundEqualsValue(initializer, expressionInitializer), boundInitializer)
```

Comment:
I think it more likely that flow analysis aborts if given a bad expression, but that feeding flow analysis a BoundEqualsValue that isn't bad but that has a bad Value causes the analysis to occur (and get an assertion failure) for a bad lambda.

There may be a few places that need to be taught to skip through BoundEqualsValue nodes. The first approach--adding some extra entries in the map from syntax nodes to bound nodes--was actually more robust in that it could not introduce problems for other kinds of analysis.